maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java

```diff
@@ -86,27 +86,29 @@ public FinancialAssessmentEntity update(FinancialAssessmentDTO financialAssessme
             updateChildWeightings(financialAssessment, existingAssessment);
         }
 
-
+        // Retrieve the list of existing income evidence records from the existing assessment
         List<FinAssIncomeEvidenceEntity> existingIncomeEvidences = existingAssessment.getFinAssIncomeEvidences();
-        for (FinAssIncomeEvidenceEntity finAssIncomeEvidenceEntity : existingIncomeEvidences) {
-            log.info("Existing evidence: " + finAssIncomeEvidenceEntity.getIncomeEvidence());
-            if (financialAssessment.getFinAssIncomeEvidences()
-                    .stream()
-                    .noneMatch(evidenceDTO -> evidenceDTO.getIncomeEvidence()
-                            .equals(finAssIncomeEvidenceEntity.getIncomeEvidence()))) {
-                log.info("Deleting Financial Assessment Income Evidence Id : " + finAssIncomeEvidenceEntity.getId());
-                finAssIncomeEvidenceRepository.deleteById(finAssIncomeEvidenceEntity.getId());
+        existingIncomeEvidences.forEach(evidence -> {
+            log.info("Existing evidence: {}", evidence.getIncomeEvidence());
+            // Check if the current evidence is not present in the updated financial assessment
+            if (financialAssessment.getFinAssIncomeEvidences().stream()
+                    .noneMatch(dto -> dto.getIncomeEvidence().equals(evidence.getIncomeEvidence()))) {
+                log.info("Deleting Financial Assessment Income Evidence Id: {}", evidence.getId());
+                finAssIncomeEvidenceRepository.deleteById(evidence.getId());
             }
-        }
+        });
```

Comment:
I think we can remove this code as:
`existingAssessment.getFinAssIncomeEvidences().clear()` should remove all existing records.

Reply:
Clear, will only clear the list and saveAndFlush won't delete the records. To simplify further added delete based on FIAS_ID to delete all the records as against deleting based on individual income evidence id

Reply:
AFAIK we just use `clear()` in the `HardshipReviewImpl` and it seems to work there?

Reply:
Did it clear the delta(removed the records) from the table? I did test it and didn't seem to work. May be will try again